server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -511,11 +511,26 @@ private static void ensureRemoteStoreNodesCompatibility(DiscoveryNode joiningNod
         assert existingNodes.isEmpty() == false;
 
         CompatibilityMode remoteStoreCompatibilityMode = REMOTE_STORE_COMPATIBILITY_MODE_SETTING.get(metadata.settings());
+        List<String> reposToSkip = new ArrayList<>();
+        // Skip checking for remote routing table repo if feature is not enabled.
+        if (!RemoteStoreNodeAttribute.isRemoteRoutingTableEnabled(metadata.settings())) {
+            String joiningNodeRepoName = joiningNode.getAttributes()
+                .get(RemoteStoreNodeAttribute.REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY);
+            String existingNodeRepoName = existingNodes.get(0)
+                .getAttributes()
+                .get(RemoteStoreNodeAttribute.REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY);
+            if (joiningNodeRepoName != null) {
+                reposToSkip.add(joiningNodeRepoName);
+            }
+            if (existingNodeRepoName != null) {
+                reposToSkip.add(existingNodeRepoName);
+            }
+        }
         if (STRICT.equals(remoteStoreCompatibilityMode)) {
 
             DiscoveryNode existingNode = existingNodes.get(0);
             if (joiningNode.isRemoteStoreNode()) {
-                ensureRemoteStoreNodesCompatibility(joiningNode, existingNode);
+                ensureRemoteStoreNodesCompatibility(joiningNode, existingNode, reposToSkip);
```

Comment:
How does it work in MIXED mode?

Reply:
We only check for routing table repo if the feature is enabled. Since remote node check ensures we check for repos only if we expect both nodes to be on remote, we dont need explicit handling for routing table.

Reply:
Changed this to check for the repo in joiningNode only if atleast one of the existing nodes already has the repo. If no existing node has routing table repo, then we mark it to be skipped during repo check.